draft-ietf-jsonpath-base.md

```diff
@@ -253,19 +254,15 @@ Unicode Scalar Value:
   E000 to 10FFFF. JSON values of type string are sequences of Unicode scalar values.
 
 Singular Path:
-: A JSONPath expression built from segments which each produce at most one node.
+: A JSONPath expression built from segments which each, regardless of the input value,
+  produces at most one node.
 
 Selector:
 : A single item within a segment that takes the input value and produces a nodelist
   consisting of child nodes of the input value.
 
 For the purposes of this specification, a value as defined by
 {{-json}} is also viewed as a tree of nodes.
-Each node, in turn, holds a value.
```

Comment:
Where is this snippet going?  E.g., saying that out of the members of a map, the values are nodes and the keys are not, seems important to me.

Reply:
The following definitions are the crucial ones and make it clear which are the values of an object:

> Children (of a node): If the node is an array, each of its elements, or if the node is an object, each of its member values (but not its member names). If the node is neither an array nor an object, it has no children.

> Descendants (of a node): The children of the node, together with the children of its children, and so forth recursively. More formally, the descendants relation between nodes is the transitive closure of the children relation.

Since we don't use the term "tree" elsewhere in the spec, I'll delete the above sentence.

Reply:
In these definitions, children and descendants are values, not nodes.  Maybe we can clarify this.

Otherwise, I agree that the deleted text is indeed covered.

Reply:
I've clarified that children (and therefore descendants) are nodes rather than values.

I've also moved a statement out of the terminology section which didn't belong there.